### ISSUE TITLE
FP bool_comparison: json

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#3703]
`clippy 0.0.212 (f175352 2019-01-24)`

````rust
use serde_json::*;

fn main() {
    let object = json!({
        "A": ["a", "á", "à"],
    });

    if object["B"][0] == true {}
}
````
warns
````
warning: equality checks against true are unnecessary
 --> src/main.rs:8:8
  |
8 |     if object["B"][0] == true {}
  |        ^^^^^^^^^^^^^^^^^^^^^^ help: try simplifying it as shown: `object["B"][0]`
  |
  = note: #[warn(clippy::bool_comparison)] on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#bool_comparison
````
however 
````rust
if object["B"][0] {}
````
does not work:
````
error[E0308]: mismatched types
 --> src/main.rs:8:8
  |
8 |     if object["B"][0] {}
  |        ^^^^^^^^^^^^^^ expected bool, found enum `serde_json::value::Value`
  |
  = note: expected type `bool`
             found type `serde_json::value::Value`

error: aborting due to previous error
````

**Comment by phansch** [comment]
We should probably return early if we can detect that the compared value has its own implementation of `PartialEq`, like serde_json: https://github.com/serde-rs/json/blob/master/src/value/partial_eq.rs#L93
